test/unit/os/fs_spec.lua

```diff
@@ -91,31 +91,34 @@ describe('fs function', function()
   end)
 
   describe('os_dirname', function()
-    local length
-
-    local function os_dirname(buf, len)
-      return fs.os_dirname(buf, len)
-    end
-
-    before_each(function()
-      length = (string.len(lfs.currentdir())) + 1
-      buffer = cstr(length, '')
+    itp('returns OK and writes current directory to the buffer', function()
+      local length = string.len(lfs.currentdir()) + 1
+      local buf = cstr(length, '')
+      eq(OK, fs.os_dirname(buf, length))
+      eq(lfs.currentdir(), ffi.string(buf))
     end)
 
-    itp('returns OK and writes current directory into the buffer if it is large\n    enough', function()
-      eq(OK, (os_dirname(buffer, length)))
-      eq(lfs.currentdir(), (ffi.string(buffer)))
+    itp('returns FAIL if the buffer is too small', function()
+      local length = string.len(lfs.currentdir()) + 1
+      local buf = cstr(length - 1, '')
+      eq(FAIL, fs.os_dirname(buf, length - 1))
     end)
+  end)
 
-    -- What kind of other failing cases are possible?
-    itp('returns FAIL if the buffer is too small', function()
-      local buf = cstr((length - 1), '')
-      eq(FAIL, (os_dirname(buf, (length - 1))))
+  describe('os_chdir', function()
+    itp('accepts "~" path', function()
+      local homedir = os.getenv('HOME')
+      local length = 4096
+      local buf = cstr(length, '')
+      -- XXX: os_chdir returns 0 for success, not OK (1). Sigh...
+      eq(0, fs.os_chdir('~'))
```

Comment:
This succeeds on my system. It fails (as our code expects) on quickbuild and travis CI.